maven-embedder/src/main/java/org/apache/maven/cli/props/MavenPropertiesLoader.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.cli.props;
+
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.StringTokenizer;
+import java.util.function.Function;
+
+import static org.apache.maven.cli.props.InterpolationHelper.substVars;
+
+public class MavenPropertiesLoader {
+
+    public static final String INCLUDES_PROPERTY = "${includes}"; // includes
+
+    public static final String OVERRIDE_PREFIX =
+            "maven.override."; // prefix that marks that system property should override defaults.
+
+    public static void loadProperties(
+            java.util.Properties properties, Path path, Function<String, String> callback, boolean escape)
+            throws Exception {
+        MavenProperties sp = new MavenProperties(false);
+        try {
+            sp.load(path);
+        } catch (NoSuchFileException ex) {
```

Comment:
Should we check also other IOException or path exist also here?

Reply:
I've added a check for the existence of the file and let the exception bubble up. The code is nicer.